zebra-chain/src/transaction/serialize.rs

```diff
@@ -672,6 +672,53 @@ impl ZcashSerialize for Transaction {
                 // `proofsOrchard`, `vSpendAuthSigsOrchard`, and `bindingSigOrchard`.
                 orchard_shielded_data.zcash_serialize(&mut writer)?;
             }
+
+            #[cfg(feature = "tx_v6")]
+            Transaction::V6 {
+                network_upgrade,
+                lock_time,
+                expiry_height,
+                inputs,
+                outputs,
+                sapling_shielded_data,
+                orchard_shielded_data,
+            } => {
+                // Transaction V5 spec:
+                // https://zips.z.cash/protocol/protocol.pdf#txnencoding
+
+                // Denoted as `nVersionGroupId` in the spec.
+                writer.write_u32::<LittleEndian>(TX_V5_VERSION_GROUP_ID)?;
+
+                // Denoted as `nConsensusBranchId` in the spec.
+                writer.write_u32::<LittleEndian>(u32::from(
+                    network_upgrade
+                        .branch_id()
+                        .expect("valid transactions must have a network upgrade with a branch id"),
+                ))?;
+
+                // Denoted as `lock_time` in the spec.
+                lock_time.zcash_serialize(&mut writer)?;
+
+                // Denoted as `nExpiryHeight` in the spec.
+                writer.write_u32::<LittleEndian>(expiry_height.0)?;
+
+                // Denoted as `tx_in_count` and `tx_in` in the spec.
+                inputs.zcash_serialize(&mut writer)?;
+
+                // Denoted as `tx_out_count` and `tx_out` in the spec.
+                outputs.zcash_serialize(&mut writer)?;
+
+                // A bundle of fields denoted in the spec as `nSpendsSapling`, `vSpendsSapling`,
+                // `nOutputsSapling`,`vOutputsSapling`, `valueBalanceSapling`, `anchorSapling`,
+                // `vSpendProofsSapling`, `vSpendAuthSigsSapling`, `vOutputProofsSapling` and
+                // `bindingSigSapling`.
+                sapling_shielded_data.zcash_serialize(&mut writer)?;
+
+                // A bundle of fields denoted in the spec as `nActionsOrchard`, `vActionsOrchard`,
+                // `flagsOrchard`,`valueBalanceOrchard`, `anchorOrchard`, `sizeProofsOrchard`,
+                // `proofsOrchard`, `vSpendAuthSigsOrchard`, and `bindingSigOrchard`.
+                orchard_shielded_data.zcash_serialize(&mut writer)?;
```

Comment:
```suggestion
                orchard_shielded_data.zcash_serialize(&mut writer)?;
                
                // TODO: Add the rest of v6 transaction fields.
```

We can be more clear that some fields are missing.